src/storage/src/server.rs

```diff
@@ -9,38 +9,38 @@
 
 //! An interactive dataflow server.
 
-use std::sync::{Arc, Mutex};
+use std::sync::Arc;
+use std::thread::Thread;
 
-use anyhow::anyhow;
 use timely::communication::initialize::WorkerGuards;
-use tokio::sync::mpsc;
 
-use mz_ore::metrics::MetricsRegistry;
+use mz_cluster::server::TimelyContainerRef;
 use mz_ore::now::NowFn;
 use mz_persist_client::cache::PersistClientCache;
-use mz_service::local::LocalClient;
 use mz_storage_client::client::StorageClient;
+use mz_storage_client::client::{StorageCommand, StorageResponse};
 use mz_storage_client::types::connections::ConnectionContext;
+use timely::worker::Worker as TimelyWorker;
 
 use crate::sink::SinkBaseMetrics;
 use crate::source::metrics::SourceBaseMetrics;
 use crate::storage_state::Worker;
 use crate::DecodeMetrics;
 
 /// Configures a dataflow server.
+#[derive(Clone)]
 pub struct Config {
-    /// The number of worker threads to spawn.
-    pub workers: usize,
-    /// The Timely configuration
-    pub timely_config: timely::Config,
     /// Function to get wall time now.
     pub now: NowFn,
-    /// Metrics registry through which dataflow metrics will be reported.
-    pub metrics_registry: MetricsRegistry,
     /// Configuration for source and sink connection.
     pub connection_context: ConnectionContext,
-    /// `persist` client cache.
-    pub persist_clients: Arc<PersistClientCache>,
+
+    /// Metrics for sources.
+    pub source_metrics: SourceBaseMetrics,
```

Comment:
It's nice how these `Configs` for compute and storage are cleaned up!

Reply:
yes!